reference/conanfile/methods/build_requirements.rst

```diff
@@ -27,6 +27,45 @@ The ``tool_requires`` is equivalent to ``requires()`` with the following traits:
 - ``headers=False`` A tool requirement does not have headers.
 - ``libs=False``: A tool requirement does not have libraries to be linked by the consumer (if it had libraries they would be in the "build" context and could be incompatible with the "host" context of the consumer package). 
 
+<host_version>
+**************
+
+.. include:: ../../../common/experimental_warning.inc
+
+This syntax is useful when you're using the same package recipe as a *requires* and as a *tool_requires* and you want to avoid
+conflicting downstream if any user decides to override the original *requires* version, i.e., the user could end up with
+two different versions in host and build context of the same dependency.
```

Comment:
I would consider adding something along the lines of:

"The `<host_version>` specifier allows us to ensure that the version resolved for the `tool_requires` always matches the one for the host requirement.